pkg/scheduler/scheduler.go

```diff
@@ -123,6 +133,10 @@ func (s *Scheduler) schedule(ctx context.Context) {
 		}
 		if s.queues.RequeueWorkload(ctx, &w) {
 			log.V(2).Info("Workload re-queued", "queuedWorkload", klog.KObj(w.Obj), "queue", klog.KRef(w.Obj.Namespace, w.Obj.Spec.QueueName))
+			if err := workload.UpdateWorkloadStatus(ctx, s.client, w.Obj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse,
+				"Workload re-queued", "Pending"); err != nil {
+				log.Error(err, "Updating QueuedWorkload status")
+			}
 		}
```

Comment:
encapsulate this (the call to requeue and updating the condition) in a function which we can use here and inside the admit function in the same place were we requeue (https://github.com/kubernetes-sigs/kueue/blob/228e329a424369c0de84bdc61af180ce3869a0c3/pkg/scheduler/scheduler.go#L252)

The function would accept a message, inside admit we pass err.Error(), while here we pass "workload didn't fit"; in a followup, the `nominate` func should return two lists, the entries that fit and the ones that didn't, and for the latter list each entry should include a reason why it didn't that we can pass as the condition message.

Reply:
done